ffi/diplomat/src/data_struct.rs

```diff
@@ -0,0 +1,88 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use alloc::borrow::Cow;
+use alloc::string::String;
+
+#[diplomat::bridge]
+pub mod ffi {
+    use super::str_to_cow;
+    use alloc::boxed::Box;
+    use diplomat_runtime::DiplomatResult;
+    use icu_decimal::provider::{
+        AffixesV1, DecimalSymbolsV1, DecimalSymbolsV1Marker, GroupingSizesV1,
+    };
+    use icu_provider::dynutil::UpcastDataPayload;
+    use icu_provider::erased::ErasedDataStructMarker;
+    use icu_provider::prelude::DataPayload;
+
+    #[diplomat::opaque]
+    /// A generic data struct to be used by ICU4X
+    ///
+    /// This can be used to construct a StructDataProvider.
+    pub struct ICU4XDataStruct(pub(crate) Option<DataPayload<ErasedDataStructMarker>>);
+
+    impl ICU4XDataStruct {
+        /// Construct a new DecimalSymbolsV1 data struct.
+        ///
+        /// See the [rust docs](https://unicode-org.github.io/icu4x-docs/doc/icu/decimal/provider/struct.DecimalSymbolsV1.html) for more details.
+        #[allow(clippy::too_many_arguments)]
+        pub fn create_decimal_symbols(
```

Comment:
Nit: I would strongly prefer a struct with all these fields as opposed to a function with 10 arguments.

Reply:
Putting `&str` inside structs is going to be problematic: structs have bidirectional flow (can be both inputs and outputs), whereas arguments are unidirectional (just inputs), and the lifetimes won't be handled right. I'm staying within the set of features diplomat supports well, lifetimes aren't one of them.

Reply:
ok, please change it to `create_decimal_symbols_v1` then (name the function the same as the struct).

Reply:
Thanks for changing `try_new_from_decimal_symbols_v1`, but please also change `create_decimal_symbols` to `create_decimal_symbols_v1`